### PR TITLE
add `Picos.Computation.peek_exn`

### PR DESCRIPTION
**Pull request by c-cube**
`peek` is often used in the fast path of some operations, so
ideally it should not allocate.

### CODE REVIEW

File: lib/picos/ocaml5/picos_ocaml.ml [hunk 1]
```diff
@@ -63,7 +63,7 @@ module Handler = struct
       Some
         (fun k ->
           match h.yield c with
-          | unit -> Effect.Deep.continue k unit
+          | () -> Effect.Deep.continue k ()
```
Comment (polytypic) [inline lib/picos/ocaml5/picos_ocaml.ml:66]: This was actually intentional (potentially generating a shorter instruction, i.e. a register to register move), but it doesn't really matter. 😄

**Review by polytypic — APPROVED** [review]
I added a commit to reformat the doc comments and added comments on the use of `inline always`.

LGTM, thanks for the contribution!

I'll squash merge shortly.

File: lib/picos/ocaml5/picos_ocaml.ml (hunk 1 above)
Comment (c-cube) [reply lib/picos/ocaml5/picos_ocaml.ml:66]: hu, does the codegen change?? :cold_sweat: I expect a match against `()` to be fully skipped…

**Comment by c-cube** [comment]
Thanks! :grin:

**Pull request merged**